pyarlo/base_station.py

```diff
@@ -19,11 +24,144 @@ def __init__(self, name, attrs, arlo_session):
 
+    def thread_function(self):
+        """Thread function."""
+        url = SUBSCRIBE_ENDPOINT + "?token=" + self._session_token
+        data = self._session.query(url, method='GET', raw=True, stream=True)
+        self.__sseclient = sseclient.SSEClient(data)
+        if self.__sseclient:
+            self.__subscribed = True
+        for event in (self.__sseclient).events():
+            if not self.__subscribed:
+                break
+            data = json.loads(event.data)
+            if 'status' in data:
+                if data['status'] == "connected":
+                    _LOGGER.debug("Successfully subscribed this base station")
+            elif 'action' in data:
+                action = data['action']
+                if action == "logout":
+                    _LOGGER.debug("Logged out by some other entity")
+                    self.__subscribed = False
+                    break
+                elif action == "is":
+                    self.__events.append(data)
+
+    def _get_event_stream(self):
+        """Spawn a thread and monitor the Arlo Event Stream."""
+        event_thread = threading.Thread(target=self.thread_function)
+        event_thread.start()
+
+    def _subscribe_myself(self):
+        """Subscribe this base station for all events."""
+        return self.__run_action(
+            method='SET',
+            resource='subscribe',
+            mode=None,
+            publish_response=False)
+
+    def _unsubscribe_myself(self):
+        """Unsubscribe this base station for all events."""
+        url = UNSUBSCRIBE_ENDPOINT
+        return self._session.query(url, method='GET', raw=True, stream=False)
+
+    def _close_event_stream(self):
+        """Stop the Event stream thread."""
+        self.__subscribed = False
+
+    def publish_and_get_event(self, resource):
+        """Publish and get the event from base station."""
+        self._get_event_stream()
+        self._subscribe_myself()
+
+        this_event = ''
+        status = self.__run_action(
+            method='GET',
+            resource=resource,
+            mode=None,
+            publish_response=False)
+        if status == 'success':
+            for i in range(0, 5):
+                _LOGGER.debug("Trying instance " + str(i))
+                for event in self.__events:
+                    if event['resource'] == resource:
+                        this_event = event
+                        self.__events.remove(event)
+                        break
+                if this_event:
+                    break
+                else:
+                    time.sleep(1)
+
+        self._unsubscribe_myself()
+        self._close_event_stream()
+
+        if this_event:
+            return this_event
+
+        return None
+
+    def __run_action(
+            self,
+            method='GET',
+            resource=None,
+            mode=None,
+            publish_response=None):
+        """Run action."""
```

Comment:
@viswa-swami  I know that the older method did not have the doc string, but could you include the doc string now?